### PR TITLE
[Logs]: Added a config var to enable TCP forward over SSL on port 443

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Added a config var to enable TCP forward over SSL on port 443

### Motivation

Enable customers to send logs on default SSL port.

### CODE REVIEW

File: pkg/logs/agent.go [hunk 1]
```diff
@@ -38,12 +38,19 @@ func NewAgent(sources *config.LogSources) *Agent {
 	messageChan := make(chan message.Message, config.ChanSize)
 	auditor := auditor.New(messageChan, config.LogsAgent.GetString("logs_config.run_path"))
 
+	var serverConfig sender.ServerConfig
+	switch {
+	case LogsAgent.GetBool("logs_config.use_port_443"):
+		serverConfig := sender.SSLServerConfig(config.LogsAgent.GetString("logs_config.dd_url"))
```
Comment (tmichelet) [inline pkg/logs/agent.go]: we'll want to use a specific hardcoded dns when this flag is set

**Review by tmichelet — APPROVED** [review]

**Comment by tmichelet** [comment]
thanks @ajacquemot !

**Review by arbll — APPROVED** [review]
:shipit:

**Review by olivielpeau — APPROVED** [review]
LGTM 👌 

I'd be interested in reviewing the public-facing docs too (when they're ready)

**Comment by tmichelet** [comment]
@olivielpeau @NBParis to confirm but I would like us not to document this publicly, this is a workaround we put in place that we will suggest under certain well defined conditions

**Pull request merged**